File: app/src/main/java/com/sxt/chat/activity/SelectAgeActivity.java
package com.sxt.chat.activity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.sxt.chat.R;
import com.sxt.chat.base.HeaderActivity;
import com.sxt.chat.db.User;
import com.sxt.chat.utils.glide.GlideCircleTransform;
import com.sxt.chat.view.picker.ScrollPickerView;
import com.sxt.chat.view.picker.StringScrollPicker;

import java.util.ArrayList;
import java.util.List;

import cn.bmob.v3.BmobUser;

/**
 * Created by izhaohu on 2018/1/26.
 */

public class SelectAgeActivity extends HeaderActivity implements View.OnClickListener {

    private TextView select;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_select_age);
        setTitle(R.string.age);
        final StringScrollPicker scrollPicker = (StringScrollPicker) findViewById(R.id.scrollPicker);
        select = (TextView) findViewById(R.id.select);
        TextView next = (TextView) findViewById(R.id.next);
        next.setOnClickListener(this);
        List<CharSequence> ages = new ArrayList<>();
        for (int i = 0; i < 120; i++) {
            ages.add(String.valueOf(i));
        }
        scrollPicker.setData(ages);
        scrollPicker.setOnSelectedListener(new ScrollPickerView.OnSelectedListener() {
            @Override
            public void onSelected(ScrollPickerView scrollPickerView, int position) {
                select.setText(scrollPicker.getSelectedItem());
            }
        });
        select.setText(scrollPicker.getSelectedItem());

        updateHeadPortrait();
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.next:
                String trim = select.getText().toString().trim();
                if (TextUtils.isEmpty(trim)) {
                    Toast("请选择年龄");
                    return;
                }
                Intent intent = new Intent();
                intent.putExtra(String.valueOf(BasicInfoActivity.REQUESTCODE_AGE), trim);
                setResult(RESULT_OK, intent);
                finish();
        }
    }

    private void updateHeadPortrait() {
        User user = BmobUser.getCurrentUser(User.class);
        if (user != null) {
            if ("M".equals(user.getGender())) {
                update(user.getImgUri(), R.mipmap.men);
            } else {
                update(user.getImgUri(), R.mipmap.female);
            }
        }
    }

    private void update(String url, int placeHolder) {
        Glide.with(this)
                .load(url)
                .error(placeHolder)
                .bitmapTransform(new GlideCircleTransform(this))
//                .skipMemoryCache(true)//跳过内存
//                .diskCacheStrategy(DiskCacheStrategy.NONE)//想要生效必须添加 跳过内存
                .into((ImageView) findViewById(R.id.img));
    }
}

File: app/src/main/java/com/sxt/chat/activity/SelectAgeActivity.kt
package com.sxt.chat.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import cn.bmob.v3.BmobUser
import com.bumptech.glide.Glide
import com.sxt.chat.R
import com.sxt.chat.base.HeaderActivity
import com.sxt.chat.db.User
import com.sxt.chat.utils.glide.GlideCircleTransformer
import com.sxt.chat.view.picker.StringScrollPicker
import java.util.*

/**
 * Created by izhaohu on 2018/1/26.
 */

class SelectAgeActivity : HeaderActivity(), View.OnClickListener {

    private var select: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_select_age)
        setTitle(R.string.age)
        val scrollPicker = findViewById<StringScrollPicker>(R.id.scrollPicker)
        select = findViewById<TextView>(R.id.select)
        val next = findViewById<TextView>(R.id.next)
        next.setOnClickListener(this)
        val ages = ArrayList<CharSequence>()
        for (i in 0..119) {
            ages.add(i.toString())
        }
        scrollPicker.data = ages
        scrollPicker.setOnSelectedListener { _, _ -> select!!.text = scrollPicker.selectedItem }
        select!!.text = scrollPicker.selectedItem
        updateHeadPortrait()
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.next -> {
                val trim = select!!.text.toString().trim()
                val intent = Intent()
                intent.putExtra("${BasicInfoActivity.REQUESTCODE_AGE}", trim)
                setResult(RESULT_OK, intent)
                finish()
            }
        }
    }

    private fun updateHeadPortrait() {
        val user = BmobUser.getCurrentUser(User::class.java)
        if (user != null) {
            if ("M" == user.gender) {
                update(user.imgUri, R.mipmap.men)
            } else {
                update(user.imgUri, R.mipmap.women)
            }
        }
    }

    private fun update(url: String, placeHolder: Int) {
        Glide.with(this)
                .load(url)
                .error(placeHolder)
                .bitmapTransform(GlideCircleTransformer(this))
                //                .skipMemoryCache(true)//跳过内存
                //                .diskCacheStrategy(DiskCacheStrategy.NONE)//想要生效必须添加 跳过内存
                .into(findViewById<ImageView>(R.id.img))
    }
}
